cpp/src/io/utilities/datasource.cpp

```diff
@@ -227,14 +228,27 @@ class memory_mapped_source : public file_source {
   }
 
  private:
+  [[nodiscard]] bool should_register_mmap_buffer()
+  {
+    if (_map_addr == nullptr) { return false; }
+
+    auto const policy = getenv_or("LIBCUDF_MMAP_REGISTER_ENABLED", std::string{"AUTO"});
```

Comment:
Just a comment: In a future improvement, boolean type handling could be made more flexible by allowing both letter cases and supporting common key words such as "yes", "true", similar to what KvikIO does [here](https://github.com/rapidsai/kvikio/blob/branch-24.12/cpp/include/kvikio/defaults.hpp#L49).